src/components/atom_word_cards_frame_surfing_button/index.tsx

```diff
@@ -2,16 +2,21 @@ import StyledIconButtonAtom from '@/atoms/StyledIconButton'
 import SurfingIcon from '@mui/icons-material/Surfing'
 import { FC } from 'react'
 import { useSemesterClick } from '@/hooks/semesters/use-semester-click.hook'
-import { useRecoilCallback, useRecoilValue } from 'recoil'
-import { semestersState } from '@/recoil/words/semesters.state'
+import { useRecoilCallback, useRecoilState, useRecoilValue } from 'recoil'
+import {
+  isSemesterExpandedState,
+  semestersState,
+} from '@/recoil/words/semesters.state'
 import { selectedSemesterSelector } from '@/recoil/words/tags.selectors'
 
 const WordCardsFrameSurfingButton: FC = () => {
   const [, onSemesterClick] = useSemesterClick()
+  const [, setSemesterExpanded] = useRecoilState(isSemesterExpandedState)
 
   const onClick = useRecoilCallback(
     ({ snapshot }) =>
       async () => {
+        setSemesterExpanded(true)
```

Comment:
Nice. But you can also set data like the following:
https://github.com/ajktown/wordnote/blob/b12beba84ca4c530e845d0324ef066721c952fb4/src/hooks/auth/use-auth-prep.hook.ts#L8-L13

Give a shot!

Reply:
Also make sure to run the `setSemesterExpanded` 
after
`if (semesters === undefined || semesters.length <= 1) return`
just in case.

Can you guess why?

Reply:
handled